Mathlib/Order/SuccPred/Arithmetic.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Algebra.Order.ZeroLEOne
+import Mathlib.Data.Int.Cast.Defs
+import Mathlib.Order.SuccPred.Basic
+
+/-!
+# Interaction between successors and arithmetic
+
+We define the `SuccAddOrder` and `PredSubOrder` typeclasses, for orders satisfying `succ x = x + 1`
+and `pred x = x - 1` respectively. This allows us to transfer the API for successors and
+predecessors into these common arithmetical forms.
+-/
+
+/-- A typeclass for `succ x = x + 1`. -/
+class SuccAddOrder (α : Type*) [Preorder α] [Add α] [One α] extends SuccOrder α where
+  succ_eq_add_one (x : α) : succ x = x + 1
+
+/-- A typeclass for `pred x = x - 1`. -/
+class PredSubOrder (α : Type*) [Preorder α] [Sub α] [One α] extends PredOrder α where
+  pred_eq_sub_one (x : α) : pred x = x - 1
+
+variable {α : Type*} {x y : α}
+
+namespace Order
+
+section Preorder
+
+variable [Preorder α]
+
+theorem succ_eq_add_one [Add α] [One α] [SuccAddOrder α] (x : α) : succ x = x + 1 :=
+  SuccAddOrder.succ_eq_add_one x
+
+theorem pred_eq_sub_one [Sub α] [One α] [PredSubOrder α] (x : α) : pred x = x - 1 :=
+  PredSubOrder.pred_eq_sub_one x
+
+theorem add_one_le_of_lt [Add α] [One α] [SuccAddOrder α] (h : x < y) : x + 1 ≤ y := by
+  rw [← succ_eq_add_one]
+  exact succ_le_of_lt h
+
+theorem le_sub_one_of_lt [Sub α] [One α] [PredSubOrder α] (h : x < y) : x ≤ y - 1 := by
+  rw [← pred_eq_sub_one]
+  exact le_pred_of_lt h
+
+theorem add_one_le_iff_of_not_isMax [Add α] [One α] [SuccAddOrder α] (hx : ¬ IsMax x) :
+    x + 1 ≤ y ↔ x < y := by
+  rw [← succ_eq_add_one, succ_le_iff_of_not_isMax hx]
+
+theorem le_sub_one_iff_of_not_isMin [Sub α] [One α] [PredSubOrder α] (hy : ¬ IsMin y) :
+    x ≤ y - 1 ↔ x < y := by
+  rw [← pred_eq_sub_one, le_pred_iff_of_not_isMin hy]
+
+theorem add_one_le_iff [Add α] [One α] [SuccAddOrder α] [NoMaxOrder α] : x + 1 ≤ y ↔ x < y :=
+  add_one_le_iff_of_not_isMax (not_isMax x)
+
+theorem le_sub_one_iff [Sub α] [One α] [PredSubOrder α] [NoMinOrder α] : x ≤ y - 1 ↔ x < y :=
+  le_sub_one_iff_of_not_isMin (not_isMin y)
+
+@[simp]
+theorem succ_zero [AddZeroClass α] [One α] [SuccAddOrder α] : succ (0 : α) = 1 := by
+  rw [succ_eq_add_one, zero_add]
+
+@[simp]
+theorem pred_zero [SubNegMonoid α] [One α] [PredSubOrder α] : pred (0 : α) = -1 := by
```

Comment:
There's some theorems like these that really only apply to `ℤ`, and which won't be very useful anyways since the simp-normal form of `Order.pred` for ℤ is `Int.pred`. Do I just go ahead and remove them? Or do we want them for symmetry?